velox/functions/sparksql/String.h

```diff
@@ -218,6 +218,25 @@ struct Sha2HexStringFunction {
   }
 };
 
+/// space function
+/// space(int) -> string
+/// Returns a string consisting of `count` spaces.
+template <typename T>
+struct SpaceFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const int32_t& count) {
+    if (count <= 0) {
+      result.setEmpty();
+      return;
+    }
+    result.resize(count);
```

Comment:
There needs to be some protection against too large value for the 'count'. What's the limit in Spark? If no limit, well have to introduce our own.

Reply:
> There needs to be some protection against too large value for the 'count'. What's the limit in Spark? If no limit, well have to introduce our own.

Spark has no limit. If it is executed by Spark and the count is close to Int.Max_Value， e.g 2147483600, the data will be returned if the memory is enough; if it is Int.Max_Value, the jvm will not support: Requested array size exceeds VM limit.
If we define a limit, the execution results of spark may deviate. what do you think is the appropriate limit?

Reply:
Let's set 1MB (2^20) limit. Should be large enough for all reasonable queries.